### PR TITLE
mgr/{restful,zabbix}: remove deprecated module

### PR DESCRIPTION
**Pull request by epuertat**
Context:
* `restful` deprecation has been discussed multiple times at CLT meetings, and [publicly announced](https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/thread/LBKLNXH7UQL7TLFU5G52Y2SYVME4RS6P/).
* `werkzeug` dependency has this [CVE-2023-46136](https://nvd.nist.gov/vuln/detail/CVE-2023-46136), which [is not fixed in all distros](https://bugzilla.redhat.com/show_bug.cgi?id=2246314).

Detailed changes:
* Remove `restful` and `zabbix` mgr module dirs,
* Remove Python depedencies (`pecan`, `werkzeug`) from ceph.spec and debian control,
* Remove associated docs,
* Remove associated QA tests,
* Update vstart.

It also made some extra clean-ups to trailing whitespace in the affected files.

Fixes: https://tracker.ceph.com/issues/47066

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by ktdreyer** [comment]
:+1: to removing this module to reduce our dependencies.

**Review by jmolmo — APPROVED** [review]
Good search for references!! I did another one and did not find any new. it seems you have found all of them. :-D 
Let's remove the deprecated restful module!

**Comment by epuertat** [comment]
Looks like `zabbix` module also depends on `restful` 😅  but it says it's deprecated since April 2020. Once CI passes, I'll add another commit removing Zabbix module.

**Review by ktdreyer — APPROVED** [review]

**Review by tchaikov — COMMENTED** [review]
lgtm in general. could you please note this change in `PengingReleaseNotes`?

**Comment by epuertat** [comment]
> lgtm in general. could you please note this change in `PengingReleaseNotes`?

@tchaikov done!

**Review by tchaikov — APPROVED** [review]
lgtm

**Comment by epuertat** [comment]
jenkins test make check arm64

**Comment by epuertat** [comment]
jenkins test dashboard cephadm

**Comment by idryomov** [comment]
> 👍 to removing this module to reduce our dependencies.

Regarding `werkzeug`, ceph-iscsi depends on it:

https://github.com/ceph/ceph-iscsi/blob/1ba4c741f8f85bb700749b2d24f7444347ab8ba4/rbd-target-api.py#L23

**Comment by ktdreyer** [comment]
> It's not in the way of removing deprecated modules, but I wanted to mention that we still use werkzeug in at least one other place.

Thanks for mentioning this!

**Review by ktdreyer — APPROVED** [review]

**Comment by ljflores** [comment]
Added the "needs-squid-backport" label, but I will actually leave that decision to @epuertat.

If there is a backport, it can be added to the v19.1.1 milestone for the next RC: https://github.com/ceph/ceph/milestone/24

**Comment by tchaikov** [comment]
> Added the "needs-squid-backport" label, but I will actually leave that decision to @epuertat.
> 
> If there is a backport, it can be added to the v19.1.1 milestone for the next RC: https://github.com/ceph/ceph/milestone/24

why shall we backport a breaking change?

**Comment by epuertat** [comment]
jenkins test dashboard cephadm

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by ktdreyer** [comment]
@tchaikov the reason is that I don't want to maintain Flask's dependencies in the ceph container image for the lifetime of `squid`.

**Comment by epuertat** [comment]
jenkins test dashboard cephadm

**Comment by ktdreyer** [comment]
(@tchaikov, for more context, https://bugzilla.redhat.com/show_bug.cgi?id=2264275 tracks the werkzeug CVE in epel9, and the PR there has the discussion about why this is challenging to land in epel9.)

**Comment by ktdreyer** [comment]
@epuertat would you please rebase this? I think we're set to go to `main` now.

**Comment by epuertat** [comment]
jenkins retest this please

**Comment by ktdreyer** [comment]
I'm ready to merge this as soon as the tests pass.

**Comment by epuertat** [comment]
jenkins test api

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by epuertat** [comment]
jenkins test dashboard cephadm

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by epuertat** [comment]
jenkins test dashboard cephadm

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by epuertat** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
@epuertat dashboard and cephadm e2e are broken currently and we are trying to fix it here: https://github.com/ceph/ceph/pull/58068

**Comment by nizamial09** [comment]
jenkins test api

**Comment by epuertat** [comment]
@yuriw were we ok with the QA run?

@ktdreyer you can merge this as soon as Yuri responds (@nizamial09 confirmed that the CI failures are unrelated to this PR).

**Comment by yuriw** [comment]
@epuertat I don't know, I got no actions/comments on https://tracker.ceph.com/issues/66149

**Comment by epuertat** [comment]
> @epuertat I don't know, I got no actions/comments on [tracker.ceph.com/issues/66149](https://tracker.ceph.com/issues/66149)

Thanks @yuriw! I missed that one. I checked the QA output and see no errors related to the restful module.

@ktdreyer all yours, you can merge this!

**Comment by sebastian-philipp** [comment]
this is stuck right now?

**Comment by epuertat** [comment]
> this is stuck right now?

Yeah. I got tired of resolving conflicts every other week 😅 I'm surprised to see that `zabbix` and `restful` are still receiving fixes...

**Comment by epuertat** [comment]
jenkins test make check

File: PendingReleaseNotes [hunk 1]
```diff
@@ -253,6 +253,13 @@ CephFS: Disallow delegating preallocated inode ranges to clients. Config
   The 'cephfs-journal-tool --rank <fs_name>:<mds_rank> journal reset' and
   'cephfs-journal-tool --rank <fs_name>:<mds_rank> journal reset --force'
   commands require '--yes-i-really-really-mean-it'.
+* mgr/restful, mgr/zabbix: both modules, already deprecated since 2020, have been
```
Comment (idryomov) [inline PendingReleaseNotes:256]: Should the release note be moved to `>=20.0.0` section (it's still in `>=19.0.0` section)?

File: debian/ceph-mgr-modules-core.install [hunk 2]
```diff
@@ -15,8 +15,6 @@ usr/share/ceph/mgr/pg_autoscaler
 usr/share/ceph/mgr/progress
 usr/share/ceph/mgr/prometheus
 usr/share/ceph/mgr/rbd_support
-usr/share/ceph/mgr/rgw
```
Comment (idryomov) [inline debian/ceph-mgr-modules-core.install:18]: Why is this line being removed?

Comment (epuertat) [reply debian/ceph-mgr-modules-core.install:18]: No clue... Probably wrong conflict solving, as it's near the modified line below.

File: PendingReleaseNotes (hunk 1 above)
Comment (epuertat) [reply PendingReleaseNotes:256]: I'll leave up to @ktdreyer to decide if he's ok with v19 or v20?

Comment (idryomov) [reply PendingReleaseNotes:256]: Well, v19 has shipped with these modules...  I'm not sure if dropping user-facing functionality like this in a stable release is acceptable.

Comment (anthonyeleven) [reply PendingReleaseNotes:256]: Anyone using Zabbix arguably has a very loose threshold of acceptability to begin with ;)

**Comment by epuertat** [comment]
jenkins test make check

**Comment by epuertat** [comment]
`arm64` and `dashboard` tests are unrelated.

**Pull request merged**